flexmeasures/api/v3_0/sensors.py

```diff
@@ -106,8 +127,75 @@ def index(self, account: Account):
         :status 403: INVALID_SENDER
         :status 422: UNPROCESSABLE_ENTITY
         """
-        sensors = get_sensors(account=account)
-        return sensors_schema.dump(sensors), 200
+        accounts: list = [account] if account else []
+        account_ids: list = [acc.id for acc in accounts]
+
+        if all_accessible is not None:
+            consultancy_account_ids: list = [
+                acc.consultancy_account_id for acc in accounts
+            ]
+            account_ids.extend(consultancy_account_ids)
+
+        filter_statement = GenericAsset.account_id.in_(account_ids)
+
+        sensor_query = (
+            select(Sensor)
+            .join(GenericAsset, Sensor.generic_asset_id == GenericAsset.id)
+            .join(Account, GenericAsset.owner)
+            .filter(
+                or_(
+                    filter_statement,
+                    GenericAsset.account_id.is_(None),
+                )
+            )
+        )
+
+        if filter is not None:
+            sensor_query = query_sensors_by_search_terms(
+                sensor_query, search_terms=filter
+            )
+
+        if page is None:
+            sensors = db.session.scalars(sensor_query).all()
+            if all_accessible:
+                sensors = [
+                    sensor for sensor in sensors if check_access(sensor, "read") is None
+                ]
+
+            sensors_response: list = [
+                {
+                    **sensor_schema.dump(sensor),
+                    "event_resolution": naturaldelta(sensor.event_resolution),
+                }
+                for sensor in sensors
+            ]
+
+            return sensors_response, 200
+        else:
+            select_pagination: SelectPagination = db.paginate(
+                sensor_query, per_page=per_page, page=page
+            )
+            sensors = select_pagination.items
+            sensors = (
+                [sensor for sensor in sensors if check_access(sensor, "read") is None]
+                if all_accessible
+                else sensors
```

Comment:
So we only check access if `all_accessible` was set to True? Why not in any case?

Reply:
I misinterpreted the explanation on the issue as this 😅, this is now refactored and will be pushed shorlty